app/internal/statistics.py

```diff
@@ -0,0 +1,312 @@
+            min_events_in_day: minimum number of daily events the user has.
+            max_events_in_day: maximum number of daily events the user has.
+            avg_events_in_day: average number of daily events the user has.
+    """
+    events_by_date = db.query(
+        func.date(Event.start), func.count(func.date(Event.start))).join(
+        UserEvent, UserEvent.event_id == Event.id).filter(
+        UserEvent.user_id == userid).filter(
+        or_(
+            and_(Event.start >= start, Event.start <= end),
+            and_(Event.end >= start, Event.end <= end),
+            and_(Event.start <= start, Event.end >= end))).group_by(
+        func.date(Event.start)).all()
+    num_of_days_in_period = (end - start).days + 1
+    if len(events_by_date) > 0:
+        min_events_in_day = min(day[1] for day in events_by_date)
+        max_events_in_day = max(day[1] for day in events_by_date)
+        sum_events_per_period = sum(day[1] for day in events_by_date)
+    else:
+        min_events_in_day = 0
+        max_events_in_day = 0
+        sum_events_per_period = 0
+    if num_of_days_in_period > len(events_by_date):
+        min_events_in_day = 0
+    return {"day_events_stats": {"min_events_in_day": min_events_in_day,
+                                 "max_events_in_day": max_events_in_day,
+                                 "avg_events_per_day":
+                                     round(sum_events_per_period /
+                                           num_of_days_in_period, 2)}}
+
+
+def get_events_duration_statistics(
+        db: Session, userid: int, start: datetime.datetime,
+        end: datetime.datetime) -> Dict[str, Dict[str, int]]:
+    """ calculate statistics for events durations relevant for
+        the requested user and the requested date range.
+        all logic is performed in the db, while the rest is in the code.
+
+    Args:
+        db: db session.
+        userid: requested user id number for statistics.
+        start: start of date range.
+        end: end of date range.
+
+    Returns: json with
+        events_duration_statistics: (for requested date range, in minutes).
+            shortest_event: shortest event the user has.
+            longest_event: longest event the user has.
+            average_event: average event the user has.
+    """
+    events_duration_statistics = db.query(
+        (func.min(func.julianday(Event.end) - func.julianday(Event.start))
+         * NIN_IN_DAY),
+        (func.max(func.julianday(Event.end) - func.julianday(Event.start))
+         * NIN_IN_DAY),
+        (func.avg(func.julianday(Event.end) - func.julianday(Event.start))
+         * NIN_IN_DAY)
+    ).join(UserEvent, UserEvent.event_id == Event.id).filter(
+        UserEvent.user_id == userid).filter(
+        or_(
+            and_(Event.start >= start, Event.start <= end),
+            and_(Event.end >= start, Event.end <= end),
+            and_(Event.start <= start, Event.end >= end))).all()
```

Comment:
Separate to a function.
Also, the last 4 lines returns many time in your code. I'd export them to a function as well.